core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -234,10 +240,13 @@ object ConsumerPerformance {
     val numThreads = options.valueOf(numThreadsOpt).intValue
     val topic = options.valueOf(topicOpt)
     val numMessages = options.valueOf(numMessagesOpt).longValue
-    val reportingInterval = options.valueOf(reportingIntervalOpt).intValue
     val showDetailedStats = options.has(showDetailedStatsOpt)
+    val showAllStats = options.has(showAllStatsOpt)
     val dateFormat = new SimpleDateFormat(options.valueOf(dateFormatOpt))
     val hideHeader = options.has(hideHeaderOpt)
+    var reportingInterval = options.valueOf(reportingIntervalOpt).longValue
+    if(reportingInterval == -1)
+      reportingInterval = numMessages/10
```

Comment:
To avoid the `var`, this can be done as:

```
val reportingInterval = {
  val interval = options.valueOf(reportingIntervalOpt).longValue
  if (interval == -1) numMessages / 10
  else interval
}
```

It's a bit more code, but it means that you don't have to do a find to see if this is mutated elsewhere.